kedro/contrib/io/pyspark/spark_jdbc.py

```diff
@@ -144,6 +144,12 @@ def __init__(
 
         # Update properties in load_args and save_args with credentials.
         if credentials is not None:
+
+            # Check credentials for bad inputs
+            for k, v in credentials.items():
+                if v is None:
```

Comment:
Does this actually address an accidentally-missing credential? This means you'd have to specify the credential property and then leave it blank, but it doesn't do anything for properties you forgot to specify (which seems to be the more likely mistake).

Reply:
Hey, thanks for taking a look @deepyaman.

To clarify my intention: the code here was written as a guard against an exception thrown from the interface between Kedro and the JDBC.

You're right that this only catches blank credential properties, but what we found was the current blank credential property error message is quite opaque (comes back as a `NullPointerException` (see https://github.com/quantumblacklabs/kedro/issues/192)), and so I wanted to help prevent any others from sinking time into that debugging session.

While I agree that it is more likely that an individual will forget a property, I also feel that usually that gets handled by the JDBC implementation itself (Ex. "Unauthorized" for missing passwords, etc), and it may become unwieldy to reimplement all of these checks in Kedro.

Thoughts?

Thanks again; appreciate your time!

Reply:
Hey @tamsanh! Thanks for the detailed explanation. If it's an issue you and your colleague ran into, and others may as well, seems fair to me!